.ci/magician/cmd/generate_comment.go

```diff
@@ -0,0 +1,374 @@
+	}
+
+	if summary, stderr, err := r.Run(path, "git", []string{"diff", "origin/" + oldBranch, "origin/" + newBranch, "--shortstat"}, nil); err != nil {
+		return "", fmt.Errorf("error diffing %s and %s: %v\nstdout:\n%s\nstderr:\n%s", oldBranch, newBranch, err, summary, stderr)
+	} else if summary != "" {
+		return fmt.Sprintf("\n%s: [Diff](https://github.com/modular-magician/%s/compare/%s..%s) (%s)", diffTitle, repoName, oldBranch, newBranch, strings.TrimSuffix(summary, "\n")), nil
+	}
+	return "", nil
+}
+
+// Get the breaking change message including the unique tpg messages and all tpgb messages.
+func compareBreakingChanges(tpgBreaking, tpgbBreaking string) string {
+	var allMessages []string
+	if tpgBreaking == "" {
+		if tpgbBreaking == "" {
+			return ""
+		}
+		allMessages = strings.Split(tpgbBreaking, "\n")
+	} else if tpgbBreaking == "" {
+		allMessages = strings.Split(tpgBreaking, "\n")
+	} else {
+		dashExp := regexp.MustCompile("-.*")
+		tpgMessages := strings.Split(tpgBreaking, "\n")
+		tpgbMessages := strings.Split(tpgbBreaking, "\n")
+		tpgbSet := make(map[string]struct{}, len(tpgbMessages))
+		var tpgUnique []string
+		for _, message := range tpgbMessages {
+			simple := dashExp.ReplaceAllString(message, "")
+			tpgbSet[simple] = struct{}{}
+		}
+		for _, message := range tpgMessages {
+			simple := dashExp.ReplaceAllString(message, "")
+			if _, ok := tpgbSet[simple]; !ok {
+				tpgUnique = append(tpgUnique, message)
+			}
+		}
+		allMessages = append(tpgUnique, tpgbMessages...)
+	}
+	if len(allMessages) > 0 {
+		return `Breaking Change(s) Detected
+The following breaking change(s) were detected within your pull request.
+
+* ` + strings.Join(allMessages, "\n* ") + `
+
+If you believe this detection to be incorrect please raise the concern with your reviewer.
+If you intend to make this change you will need to wait for a [major release](https://www.terraform.io/plugin/sdkv2/best-practices/versioning#example-major-number-increments) window.
+An ` + "`override-breaking-change`" + `label can be added to allow merging.
+`
+	}
+	return ""
+}
+
+// Update the provider package name to the given name in the given path.
+// name should be either "old" or "new".
+func updatePackageName(name, path string, r runner) error {
+	oldPackageName := "github.com/hashicorp/terraform-provider-google-beta"
+	newPackageName := "google/provider/" + name
+	fmt.Printf("Updating package name in %s from %s to %s\n", path, oldPackageName, newPackageName)
+	if stdout, stderr, err := r.Run(path, "find", []string{".", "-type", "f", "-name", "*.go", "-exec", "sed", "-i.bak", fmt.Sprintf("s~%s~%s~g", oldPackageName, newPackageName), "{}", "+"}, nil); err != nil {
+		return fmt.Errorf("error running find: %v\nstdout:\n%s\nstderr:\n%s\n", err, stdout, stderr)
+	}
+	if stdout, stderr, err := r.Run(path, "sed", []string{"-i.bak", fmt.Sprintf("s|%s|%s|g", oldPackageName, newPackageName), "go.mod"}, nil); err != nil {
```

Comment:
Can we switch to error handling within r.Run. Or perhaps create another r.Run command that handles errors internally. It would be nice to not need to call it repeatedly. 

```
func (r *Runner) RunOrDie(path string, cmd string, args []string, env []string) string {
	if summary, stderr, err := r.Run(path, cmd, args, env); err != nil {
		log.Fatalf("error executing command: %v\nstdout:\n%s\nstderr:\n%s", err, summary, stderr)
	}
	return summary
}
```

optionally you could also absorb path where a runner is executed into some internal state managed by the runner.

r.Pushd
r.Popd
r.Setd

that way you don't need to provide the path every time.

Reply:
The issue with switching to using `log.Fatal` is that many of these commands can fail without stopping execution of the overall command. I've added a `MustRun` function for commands that cannot fail.